### PR TITLE
Update to the latest rustc (4874ca36f 2015-01-23)

### PR DESCRIPTION
**Pull request by m-r-r**
Hello,

The crate did not built with the latest version of the Rust compiler, so I tried to fix it :-)

### CODE REVIEW

**Comment by golddranks** [comment]
Yes please! This prevents Iron from building currently.

File: src/lib.rs [hunk 1]
```diff
@@ -60,9 +60,9 @@ use typeable::Typeable;
 /// `.downcast` or the `match_error!` macro. This type can be used to meet
 /// the needs of consumers like `Verifier`, but should not be used in cases
 /// where enums or generics are better suited.
-pub trait Error: Show + Send + Typeable + StdError { }
+pub trait Error: Debug + Send + Typeable + StdError { }
 
-impl<S: StdError + Show + Send + Typeable> Error for S { }
+impl<S: StdError + Debug + Send + Typeable> Error for S { }
```
Comment (reem) [inline src/lib.rs]: I'm not convinced this Debug bound is actually needed anymore, given that StdError now inherits from Display, but we can keep it for convenience for now.

File: src/lib.rs [hunk 2]
```diff
@@ -107,25 +107,34 @@ macro_rules! match_error {
...
+    impl Display for ParseError {
+        fn fmt(&self, f: &mut Formatter) -> Result<(), FmtError> {
+            self.description().fmt(f)
+        }
+    }
+
     #[test] fn test_generic() {
         fn produce_parse_error() -> Box<Error> {
-            Box::new(ParseError { location: 7u })
+            Box::new(ParseError { location: 7us })
```
Comment (reem) [inline src/lib.rs]: These hints can go away completely now.

**Comment by reem** [comment]
Thank you very much for this fix, if you could just address the nit about hinting numbers in the tests, I'll merge.

**Pull request merged**